### PR TITLE
update to node 18 and update lerna. Closes #413

### LINKED ISSUE #413: Update to node 18 and Lerna 7+
Node 14 is out of maintence in May. We've been using node 18 (`lts/hydrogen`) for most apps for a while so migrate to that. 

This also affects the lerna version

### PR DESCRIPTION
**Pull request by nick-verida**

### CODE REVIEW

**Comment by aurelticot** [comment]
It's not as simple as updating `.nvmrc`, please see details in https://github.com/verida/verida-js/issues/402

**Comment by nick-verida** [comment]
While all the thing in #402 are probably correct we've been running things under node 18 for a while (not node 20). Also for a library it seems to make sense to move to the lowest possible version so we don't lockout other projects that are on node 18.

We have to do _something_ and this seems extremely safe.

**Comment by aurelticot** [comment]
> While all the thing in #402 are probably correct we've been running things under node 18 for a while (not node 20). Also for a library it seems to make sense to move to the lowest possible version so we don't lockout other projects that are on node 18.
> 
> We have to do _something_ and this seems extremely safe.

I disagree.

This PR only updates `.nvmrc` which is only used for dev, tests and build. The `engine` requirements is what's important and what constrain other projects.

The `engine` is currently defined with `node >=14` but our tests are run with whatever is defined in `.nvmrc` so if it is node 18, there is no guarantee these libraries will work on node 14 to 17, so, no, it is not extremely safe!

It is reasonable to think no projects should run on node lower than 18 (actually even 20), so if we "have to do _something_", it is about upgrading the `engine` requirement. It's a breaking change and the purpose of #402, which highlight a broader consideration, with changes across `package.json`, `tsconfig.json` and `.nvmrc`. That's my point, it's not as simple as only updating `.nvmrc`.

**Comment by nick-verida** [comment]
> The engine is currently defined with node >=14 but our tests are run with whatever is defined in .nvmrc so if it is node 18, there is no guarantee our libraries will work on node 14 to 17. So, no, it is not extremely safe!

Yes this is a good point. But it's still better than now where the tests run in node 14 (which - again - is about to be out of maintenance)

I agree we should do all the things in #402 (although I think I'd still argue for node 18 instead of node 20). 

I don't have time to do all of them now though, and by doing this change at least we'd test under the same node version we run things on!

**Comment by nick-verida** [comment]
(I was going to update the version in package.json too but clearly we don't want to do that since we still want to allow people to use earlier versions if they choose)

**Comment by aurelticot** [comment]
> I agree we should do all the things in https://github.com/verida/verida-js/issues/402 (although I think I'd still argue for node 18 instead of node 20).

#402 suggests node 20 in `.nvmrc` **only** if we run the tests on multiple node version, otherwise `.nvmrc` should be aligned with the `engine` requirement.

> (I was going to update the version in package.json too but clearly we don't want to do that since we still want to allow people to use earlier versions if they choose)

#402 suggests the `engine` requirement to be `node >= 16`, to take into account that 14 is obsolete but 16 is still maintained and we want to support projects still using it.

> by doing this change at least we'd test under the same node version we run things on!

"... we run things on" but **we** are not the only one running things on. At this stage it's fair to say we are the main consumer of the SDK and we know who use it and can inform them there's a lack of test converage on earlier supposedly supported node version. But I personally don't want this to be a precedent, so by principle I will stay against this PR as it is. I'll leave @tahpot with the decision to merge it.

**Comment by nick-verida** [comment]
This now fixes the docs and updates lerna too

File: README.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ It's not possible to add dependencies between monorepo packages using yarn (ie:
 Unpublished dependencies between monorepo packages can be linked by:
 
 - Manually adding the expected dependency to `package.json` (ie: `@verida/encryption-utils`)
-- Run `npx lerna bootstrap` in the root directory of this project
+- Run `yarn install` in the root directory of this project
```
Comment (tahpot) [inline README.md:42]: @nick-verida This change is incorrect, please revert.

File: README.md [hunk 2]
```diff
@@ -49,7 +49,7 @@ Build everything:
 In the root directory:
 
 ```
-npx lerna bootstrap
+yarn install
```
Comment (tahpot) [inline README.md:52]: @nick-verida Is it wort adding `nvm use` here?

**Review by tahpot — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (nick-verida) [reply README.md:52]: It is in the section Developer Notes -> Node Version

File: README.md (hunk 1 above)
Comment (nick-verida) [reply README.md:42]: @tahpot yes, `lerna bootstrap` has been replaced:

> Replace lerna bootstrap with npm install (or yarn/pnpm). If you are already performing your package manager's install command somewhere in your workflow before where you had previously called lerna bootstrap, then you can just delete it instead. 

https://lerna.js.org/docs/legacy-package-management

**Review by aurelticot — CHANGES_REQUESTED** [review]
The `helpers/dist` folder has been commited by mistake

**Pull request merged**